src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -152,17 +151,12 @@ public ResultResource(ProgrammingExerciseParticipationService programmingExercis
         try {
             planKey = continuousIntegrationService.get().getPlanKey(requestBody);
         }
-        catch (BitbucketException bitbucketException) {
+        catch (ContinuousIntegrationException cISException) {
             log.error("Exception encountered when trying to retrieve the plan key from a request a new programming exercise result: {}, {} :"
-                    + "BitBucket encountered an Exception while trying to retrieve the build plan ", bitbucketException, requestBody);
+                    + "Your CIS encountered an Exception while trying to retrieve the build plan ", cISException, requestBody);
             return badRequest();
         }
-        catch (JenkinsException jenkinsException) {
-            log.error("Exception encountered when trying to retrieve the plan key from a request a new programming exercise result: {}, {}:"
-                    + " The full name of the Jenkins build plan was malformed!", jenkinsException, requestBody);
-            return badRequest();
-        }
-        catch (Exception ex) {
+        catch (Exception ex) { // Jenkins might still throw a JSONProcsessingException, the method is declared with "throws Exception" so this needs to be here as of rn
```

Comment:
Jenkins shouldn't throw the `JSONProcessingException` in my opinion. If we catch the exception in the `JenkinsService` we can adjust the signature in the CI-Service interface and get rid of the general `throws Exception` part. If you don't want to adjust it here we can also create a new issue for it.